test/asset.js

```diff
@@ -0,0 +1,267 @@
+const ERC721 = artifacts.require("TestERC721");
+const CK = artifacts.require("CryptoKittyTest");
+const ERC1155 = artifacts.require("TestERC1155");
+
+const utils = require("../utils/utilities.js");
+const { ETH_TOKEN } = require("../utils/utilities.js");
+
+const ZERO_BYTES32 = ethers.constants.HashZero;
+const ZERO_ADDRESS = ethers.constants.AddressZero;
+const SECURITY_PERIOD = 2;
+const SECURITY_WINDOW = 2;
+const LOCK_PERIOD = 4;
+const RECOVERY_PERIOD = 4;
+
+const RelayManager = require("../utils/relay-manager");
+
+contract("ArgentModule", (accounts) => {
+  let manager;
+
+  const infrastructure = accounts[0];
+  const owner = accounts[1];
+  const sender = accounts[2];
+  const recipient = accounts[4];
+  const nonceInitialiser = accounts[5];
+  const relayer = accounts[9];
+
+  let registry;
+  let transferStorage;
+  let guardianStorage;
+  let module;
+  let wallet;
+  let walletImplementation;
+  let authoriser;
+
+  let erc20;
+  let erc721;
+  let erc1155;
+  let ck;
+
+  const ckId = 0;
+  const erc721Id = 7;
+  const erc1155Id = 4;
+
+  before(async () => {
+    registry = await Registry.new();
+    guardianStorage = await GuardianStorage.new();
+    transferStorage = await TransferStorage.new();
+    authoriser = await Authoriser.new(0);
+
+    const uniswapRouter = await UniswapV2Router01.new();
+
+    module = await ArgentModule.new(
+      registry.address,
+      guardianStorage.address,
+      transferStorage.address,
+      authoriser.address,
+      uniswapRouter.address,
+      SECURITY_PERIOD,
+      SECURITY_WINDOW,
+      RECOVERY_PERIOD,
+      LOCK_PERIOD);
+
+    await registry.registerModule(module.address, ethers.utils.formatBytes32String("ArgentModule"));
+    await authoriser.addDapp(0, relayer, ZERO_ADDRESS);
+
+    walletImplementation = await BaseWallet.new();
+
+    manager = new RelayManager(guardianStorage.address, ZERO_ADDRESS);
+  });
+
+  beforeEach(async () => {
+    const proxy = await Proxy.new(walletImplementation.address);
+    wallet = await BaseWallet.at(proxy.address);
+    await wallet.init(owner, [module.address]);
+    await wallet.send(web3.utils.toWei("1"));
+    // ERC20
+    erc20 = await ERC20.new([infrastructure, sender], 10000000, 12);
+    // ERC721
+    erc721 = await ERC721.new();
+    await erc721.mint(sender, erc721Id);
+    // Crypto Kitty
+    ck = await CK.new();
+    await ck.createDumbKitty(sender);
+    // ERC1155
+    erc1155 = await ERC1155.new();
+    await erc1155.mint(sender, erc1155Id, 10000000);
+  });
+
+  function encodeTransaction(to, value, data, isTokenCall = false) {
+    return { to, value, data, isTokenCall };
+  }
+
+  async function whitelist(target) {
+    await module.addToWhitelist(wallet.address, target, { from: owner });
+    await utils.increaseTime(3);
+    const isTrusted = await module.isWhitelisted(wallet.address, target);
+    assert.isTrue(isTrusted, "should be trusted after the security period");
+  }
+
+  async function initNonce() {
```

Comment:
Can we consolidate `encodeTransaction`,  `whitelist` and `initNonce` into `utils` library. They are duplicated across all tests.